projects/plugins/jetpack/modules/search/instant-search/components/search-app.jsx

```diff
@@ -218,19 +219,25 @@ class SearchApp extends Component {
 	showResults = () => {
 		this.setState( { showResults: true } );
 		this.preventBodyScroll();
+
+		// If we've arrived via a filter or sort only, make sure the search query is set
+		if ( this.props.searchQuery === null ) {
+			this.props.setSearchQuery( '' );
+		}
 	};
 
 	hideResults = () => {
 		this.restoreBodyScroll();
 		restorePreviousHref( this.props.initialHref, () => {
 			this.setState( { showResults: false } );
+			this.props.setSearchQuery( null );
 		} );
 	};
 
-	onChangeQueryString = () => {
+	onChangeSearchQuery = () => {
 		this.getResults();
 
-		if ( this.hasActiveQuery() && ! this.state.showResults ) {
+		if ( this.props.searchQuery !== null && ! this.state.showResults ) {
 			this.showResults();
 		}
```

Comment:
Why did we have to split `onChangeQueryString` into two functions? Doesn't the new `hasActiveQuery` also perform this check?

If the `searchInputSelector` update is the issue, I think we can gate it behind a `this.props.searchQuery !== null` conditional.

Reply:
Resolved in the latest code update.